crates/ruff_python_formatter/src/string/docstring.rs

```diff
@@ -205,6 +209,10 @@ pub(crate) fn format(normalized: &NormalizedString, f: &mut PyFormatter) -> Form
         space().fmt(f)?;
     }
 
+    if is_module && trim_end.ends_with('\n') {
+        hard_line_break().fmt(f)?;
+    }
+
```

Comment:
I think this is a bug in our codebock formatting or docstring formatting over all and is not specific to module formatting (CC: @BurntSushi). 

For example:

```python
class Test:
    
    """
    Black's `Preview.module_docstring_newlines`


    A code black to format
"""
```

Black formats this to 

```python
class Test:
    """
    Black's `Preview.module_docstring_newlines`


    A code black to format
    """
```

and Ruff formats it to 

```python
class Test:
    """
    Black's `Preview.module_docstring_newlines`


    A code black to format"""
```

Note how the last line gets collapsed. 

The relevant logic is in https://github.com/astral-sh/ruff/blob/935b59adaeb7da4d0b21f524d5da46583e779e04/crates/ruff_python_formatter/src/string/docstring.rs#L388-L398

My understanding is that the last line should only be collapsed for docstrings with a single content line:

Collapse 

```python
class Test:
    """Black A code black to format
    
    """
```

Don't collapse for 

```python
class Test:
    """Black A code black to format
    d
    """
```

Keep it collapsed when it was collapsed in the source code

```python
class Test:
    """Black A code black to format
    a"""
```

I suggest that we revert this change and address this issue in its own PR

Reply:
I reverted the change just note that the current implementation  would have a different formatting if applied twice in the following example. Just flagging it so if you want we can keep this PR open until that one gets fixed.

```
"""
A docstring

.. code-block:: python

    button.on_event(ButtonClick, callback)

"""

Format once:
"""
A docstring

.. code-block:: python

    button.on_event(ButtonClick, callback)
"""

Format twice:
"""
A docstring

.. code-block:: python

    button.on_event(ButtonClick, callback)"""
```